### ISSUE TITLE
OpenBoard 1.6.1 regression: Blinking content in ternary window when advancing to next slide renders OpenBoard unresponsive

### ISSUE THREAD

**Issue by maehne** [OpenBoard-org/OpenBoard#611]
This issue might be related to issue #516, but deserves IMHO its own issue for tracking, as it is only present in OpenBoard version 1.6.1 (and maybe the 1.6.x alpha/beta releases), but was not present in OpenBoard 1.5.4: Once a ternary screen window is open and you advance in your slide set to from slide 1 to slide 2 or beyond, the content of the ternary window starts to blink between the last slide and an empty slide with the message "Processing..." in the center. At that point OpenBoard starts to cause an extremely high CPU load and becomes unresponsive to keyboard inputs until one switches with the mouse between single screen mode / multi screen mode from the menu or closes the ternary window with the help of the task/window manager. When the ternary window is blinking, the primary and secondary screen windows only show an empty slide with the message "Processing...". The behaviour can be observed on Windows, Linux, and macOS.

**Comment by maehne** [comment]
The issue is also present in OpenBoard 1.6.2.

**Comment by kaamui** [comment]
Hi,

Thank you for reporting it. "Processing..." is what is displayed before PDF can be rendered. Are you able to try on the `1.7-dev` branch ? I made some changes in this part of the code to make it compatible with real-time thumbnails, and it may have solved your issue.

**Comment by maehne** [comment]
I built branch 1.7-dev commit 47a96e1d on Debian 11. That version seems to have completely disabled the ternary screen for "Previous pages" even though the new setting allows listing three screens:

![grafik](https://user-images.githubusercontent.com/1174243/173028718-97a6311b-a71e-4761-95bb-4040af701d6a.png)

So, I cannot observe this bug on 1.7-dev, but cannot be sure whether it god fixed or just hidden. Personally, I would like to be able to enable the "Previous pages" screen in multi screen mode from case to case. Depending on the teaching activity, it may make sense or not. Maybe, this could be a button or entry in the OpenBoard menu.

**Comment by letsfindaway** [comment]
Thanks for reporting. This is a bug in the current `1.7-dev`. Yes, the display manager was completely rewritten. It should use `eDP` as "Previous page" monitor. If it does not, then this is my fault.

Unfortunately I only have two screens on my development setup, so I cannot test that and rely on testers like you.

**Comment by maehne** [comment]
@letsfindaway: Thanks for all the work your are putting into reworking the display manager!

**Comment by kaamui** [comment]
This bug is no longer present in `1.7-dev` branch.

**Comment by kaamui** [comment]
As a workaround for the 1.6.1 , 1.6.2 and 1.6.3 versions, go to Preferences -> Display tab and disable "Improve zoom execution time (can slightly affect rendering quality)"

**Comment by maehne** [comment]
@kaamui: Your workaround for 1.6.1 works for me well! Thanks!

**Comment by maehne** [comment]
@letsfindaway: Yes, Previous screen works on `1.7-dev` but not on `improve-displaymanager`. Thanks for also looking into the initial drawing of Previous screen widget issue!

**Comment by letsfindaway** [comment]
> @letsfindaway: Yes, Previous screen works on `1.7-dev` but not on `improve-displaymanager`. Thanks for also looking into the initial drawing of Previous screen widget issue!

Thanks, this did the trick! Now I know what happens and why it does not work. No more logs required! I will just need some time to make the correct sequence of events - knowing how many previous screens - creating the widgets for them - putting them at the correct position, adding the proper scenes and the same if configuration changes by user or because a monitor is disconnected... But now that I know where the problem is, I can also sort it out. Stay tuned ;)

**Comment by maehne** [comment]
@letsfindaway: Great that you found the cause. I look forward testing your improved branch once ready!

**Comment by letsfindaway** [comment]
@maehne Yep, you could give it a try. At least for me it looks better than before, but what shall I tell with just two screens ... ;)

**Comment by maehne** [comment]
@letsfindaway: I just tested commit a6cbe42d from `improve-displaymanager` branch and the Previous page monitor *works out of the box*! :-) 

I found one issue though: I can move the content on the Previous page monitor using the horizontal and vertical scroll wheels of my mouse like it is correctly possible on the Control monitor but disabled on the Display monitor. I would expect the Previous page monitor to ignore the scroll wheel inputs. The Previous page monitor also accepts pen and touch inputs, which the Display monitor ignores as well. IMHO, this should be handled in a consistent manner. Should this maybe get an own issue?

The cropping issue on the control monitor still persist.
